Base/Logic/vtkSlicerApplicationLogic.cxx

```diff
@@ -737,15 +743,46 @@ bool vtkSlicerApplicationLogic::IsPluginInstalled(const std::string& filePath,
     return false;
     }
 
+  // If the plugin is found in the application build directory (or one of its descendent), it is not
+  // considered to be installed. This is generally true unless the plugin is found in the extension
+  // install directory (or one of its descendent).
   std::string path = itksys::SystemTools::GetFilenamePath(filePath);
   std::string canonicalPath = itksys::SystemTools::GetRealPath(path.c_str());
-
   if (itksys::SystemTools::StringStartsWith(canonicalPath.c_str(), applicationHomeDir.c_str()))
     {
-    return !itksys::SystemTools::FileExists(
+    bool isAppBuildTree = itksys::SystemTools::FileExists(
           std::string(applicationHomeDir).append("/CMakeCache.txt").c_str(), true);
+    if (isAppBuildTree)
+      {
+#ifdef Slicer_BUILD_EXTENSIONMANAGER_SUPPORT
+      // Check if plugin is installed as an extension
+      {
+      std::string extensionInstallDir = itksys::SystemTools::GetRealPath(
+            applicationHomeDir + "/" Slicer_ORGANIZATION_DOMAIN "/" Slicer_EXTENSIONS_DIRBASENAME);
```

Comment:
Instead of checking if the directory is like `<applicationHome>/<org-name-or-domain>/Extensions-`, we would consolidate the approach already used in `vtkSlicerApplicationLogic::IsEmbeddedModule` by updating the signature of `vtkSlicerApplicationLogic::IsPluginInstalled` to accept a revision parameter and instead look for `Extensions-<revision>` in the plugin path.